Content.Server/Spider/SpiderSystem.cs

```diff
@@ -0,0 +1,106 @@
+
+            SubscribeLocalEvent<SpiderWebObjectComponent, ComponentStartup>(OnWebStartup);
+            SubscribeLocalEvent<SpiderComponent, ComponentStartup>(OnSpiderStartup);
+            SubscribeLocalEvent<SpiderComponent, SpiderWebActionEvent>(OnSpawnNet);
+        }
+
+        private void OnSpiderStartup(EntityUid uid, SpiderComponent component, ComponentStartup args)
+        {
+            var netAction = new InstantAction(_proto.Index<InstantActionPrototype>("SpiderWebAction"));
+            _action.AddAction(uid, netAction, null);
+        }
+
+        private void OnWebStartup(EntityUid uid, SpiderWebObjectComponent component, ComponentStartup args)
+        {
+            _appearance.SetData(uid,SpiderWebVisuals.Variant, _robustRandom.Next(1, 3));
+        }
+
+        private void OnSpawnNet(EntityUid uid, SpiderComponent component, SpiderWebActionEvent args)
+        {
+            if (args.Handled)
+                return;
+
+            var transform = Transform(uid);
+
+            if (!_mapManager.TryGetGrid(transform.GridUid, out var grid)) return;
+
+            args.Handled = true;
+
+            var coords = transform.Coordinates;
+            bool notBlocked = !IsTileBlocked(uid, coords);
+
+            if (notBlocked)
+            {
+                // Spawn web in center
+                if (!IsTileBlockedByWeb(coords))
+                    Spawn(component.WebPrototype, coords);  
+
+                // Spawn web in other directions
+                for (var i = 0; i < 4; i++)
+                {
+                    var direction = (DirectionFlag) (1 << i);
+                    coords = transform.Coordinates.Offset(direction.AsDir().ToVec());
+                    
+                    if (!IsTileBlocked(uid, coords) && !IsTileBlockedByWeb(coords))
+                        Spawn(component.WebPrototype, coords);
+                }
+            }
+        }
+
+        private bool IsTileBlocked(EntityUid user, EntityCoordinates coords)
+        {
+            foreach (var entity in coords.GetEntitiesInTile())
+            {
+                PhysicsComponent? physics = null; // We use this to check if it's impassable
+                if ((HasComp<MobStateComponent>(entity) && entity != user) || // Is it a mob?
+                    ((Resolve(entity, ref physics, false) && (physics.CollisionLayer & (int) CollisionGroup.Impassable) != 0) // Is it impassable?
+                        && !(TryComp<DoorComponent>(entity, out var door) && door.State != DoorState.Closed))) // Is it a door that's open and so not actually impassable?
+                            return true;
+            }
+            return false;
+        }
+
+        private bool IsTileBlockedByWeb(EntityCoordinates coords)
+        {
+            foreach (var entity in coords.GetEntitiesInTile())
+            {
+                if (HasComp<SpiderWebObjectComponent>(entity))
+                    return true;
+            }
+            return false;
```

Comment:
i think these are too much special casing for something that should be a much more simpler check, it makes sense that spider webs can be spawned above walls, mobs, doors and pretty much anything except other webs since they aren't a hard object like a wall
Tldr is remove `IsTileBlocked()` check and keep `IsTileBlockedByWeb()`

Reply:
Hm, spawn webs on walls? I.e. even through walls? Ok

Reply:
> Waste hours to test blocking checks
> Copy paste from mime
> Remove checks

Reply:
Oh i pointed at the mime cuz it used a GetEntitiesInTile and not the tileref thing you did before, mime does so many checks because its a wall and a hard object, it makes sense that a spider web can be on a wall

Reply:
I need be more specific about what you mean. For me it's not obvious that web can go through walls, because they are obviously solid

Reply:
Please don't copy-paste code, especially not Rane's code. If you need wall spawning logic as an action it should be generic and not require us to double any bugfixes across several hundred lines of code in 2 systems.

Reply:
Yeah the code in this review is old, it just does a single GenEntitiesInTile now so it should be fine